shared/TestStandComm/SerialTransport.h

```diff
@@ -4,25 +4,33 @@
 #include "SerialDevice.h"
 #include "Messages.h"
 
-typedef enum {
-    MSG_SEG_START    = 0,
-    MSG_SEG_ID       = 1,
-    MSG_SEG_LENGTH   = 2,
-    MSG_SEG_DATA     = 3,
-    MSG_SEG_CRC      = 4,
-    MSG_SEG_END      = 5
-} MessageSegment;
-
-typedef struct {
-    MessageSegment current_segment;
-    uint8_t bytes_read;
-    uint8_t msg_length;
-    uint16_t crc;
-} PendingMessage;
-
+/**
+ * @class SerialTransport
+ * 
+ * @brief Transport layer of the serial communication protocol
+ * 
+ * The layer is aware of "Messages" (distinct packets of data with a defined structure)
+ * and is responsible for sending and receiving entire Messages.
+ */
 class SerialTransport
 {
     private:
+        typedef enum {
+            MSG_SEG_START    = 0,
+            MSG_SEG_ID       = 1,
+            MSG_SEG_LENGTH   = 2,
+            MSG_SEG_DATA     = 3,
+            MSG_SEG_CRC      = 4,
+            MSG_SEG_END      = 5
+        } MessageSegment;
```

Comment:
Unsure if you had already mentioned in another PR if there was a specific reason that you wanted to keep the enumeration explicit by having 0,1,2,3? Otherwise I think
`typedef enum{
MSG_SEG_START, MSG_SEG_ID,....} MessageSegment;` would be ok?

Reply:
Oh yeah forgot about that. No reason to keep them here